### PR TITLE
SN258: Add project type ConversationTranslation 

### PR DESCRIPTION
**Pull request by prakharrathi25**
# Description

This PR adds a data type of Conversation and a project type of ConversationTranslation. The project type details can be found [here ](https://docs.google.com/document/d/18qdKykmwdXtX2zy_699N7spx6BBlzGknIFlAm7lvfcI/edit)

Fixes SN258

## Type of change

Please delete options that are not relevant.

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

Please describe the tests / checks that you performed to verify your changes. Provide instructions so we can reproduce. Please delete options that are not relevant.

- [x] Tested the Models using the Django Shell

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

## Checklist for Models

If you added new models / made changes to exsting models, please fill this checklist. 

- [x] The primary key of the Model is an AutoField
- [x] I have defined the `__str__` method for the Model
- [x] If the field is optional, it has been spcified explicitly
- [x] I have added a `verbose_name` for each Field.
- [x] I have added a docstring for the Model
- [x] The model is registered on Admin Site.
- [x] I have pushed the migrations.

### CODE REVIEW

File: backend/dataset/models.py [hunk 1]
```diff
@@ -275,6 +276,62 @@ def __str__(self):
...
+    prompt = models.TextField(
+        verbose_name="prompt",
+        null=True,
+        blank=True,
+        help_text=("Prompt of the conversation"),
+    )
+    speaker_count = models.IntegerField(
+        verbose_name="speaker_count",
+        null=True,
+        blank=True,
```
Comment (GokulNC) [inline backend/dataset/models.py:312]: We should not allow null values here.

File: backend/dataset/models.py [hunk 2]
```diff
@@ -275,6 +276,62 @@ def __str__(self):
...
+        help_text=("Details of the speakers involved in the conversation"),
+    )
+    language = models.CharField(
+        verbose_name="language", choices=LANG_CHOICES, max_length=15
+    )
+    conversation_json = models.JSONField(
+        verbose_name="conversation_details",
+        null=True,
+        blank=True,
```
Comment (GokulNC) [inline backend/dataset/models.py:327]: We should not allow null value here as well.

**Review by GokulNC — APPROVED** [review]
Looks good to me, thanks!

File: backend/dataset/models.py (hunk 1 above)
Comment (prakharrathi25) [reply backend/dataset/models.py:312]: Should we also disallow null values in in `speakers_json` as well?

File: backend/dataset/models.py (hunk 2 above)
Comment (prakharrathi25) [reply backend/dataset/models.py:327]: Removed.

File: backend/dataset/migrations/0030_auto_20220902_0808.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            name='conversation_json',
+            field=models.JSONField(help_text='Details of the conversation', verbose_name='conversation_details'),
+        ),
+        migrations.AlterField(
+            model_name='conversation',
+            name='speaker_count',
+            field=models.IntegerField(help_text='Number of speakers involved in conversation', verbose_name='speaker_count'),
+        ),
+        migrations.RunPython(add_default_speaker_count_where_null),
+        migrations.RunPython(add_default_organization_id_where_null),
```
Comment (prakharrathi25) [inline backend/dataset/migrations/0030_auto_20220902_0808.py:39]: @GokulNC can you review if these migrations are okay? I removed the `organization_id` `RunPython` statement so I had to add it back.

**Review by prakharrathi25 — COMMENTED** [review]
@GokulNC can you have a look at this?

**Review by ishvindersethi22 — APPROVED** [review]

**Pull request merged**